src/Symfony/Component/Notifier/Bridge/Smsc/SmscTransport.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Bridge\Smsc;
+
+use Symfony\Component\Notifier\Exception\TransportException;
+use Symfony\Component\Notifier\Exception\UnsupportedMessageTypeException;
+use Symfony\Component\Notifier\Message\MessageInterface;
+use Symfony\Component\Notifier\Message\SentMessage;
+use Symfony\Component\Notifier\Message\SmsMessage;
+use Symfony\Component\Notifier\Transport\AbstractTransport;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+class SmscTransport extends AbstractTransport
+{
+    protected const HOST = 'smsc.ru';
+
+    private $login;
+    private $password;
+    private $from;
+
+    public function __construct($username, $password, $from, HttpClientInterface $client = null, EventDispatcherInterface $dispatcher = null)
+    {
+        $this->login = $username;
+        $this->password = $password;
+        $this->from = $from;
+
+        parent::__construct($client, $dispatcher);
+    }
+
+    public function __toString(): string
+    {
+        return sprintf('smsc://%s?from=%s', $this->getEndpoint(), (string) $this->from);
+    }
+
+    public function supports(MessageInterface $message): bool
+    {
+        return $message instanceof SmsMessage;
+    }
+
+    protected function doSend(MessageInterface $message): SentMessage
+    {
+        if (!$message instanceof SmsMessage) {
+            throw new UnsupportedMessageTypeException(__CLASS__, SmsMessage::class, $message);
+        }
+
+        $query = [
+            'login' => $this->login,
+            'psw' => $this->password,
+            'sender' => (string) $this->from,
+            'phones' => $message->getPhone(),
+            'mes' => $message->getSubject(),
+            'fmt' => 3, // response as JSON
+            'charset' => 'utf-8',
+            'time' => '0-24',
+        ];
+
+        $endpoint = sprintf('https://%s/sys/send.php', $this->getEndpoint());
+        $response = $this->client->request('GET', $endpoint, ['query' => $query]);
```

Comment:
Add a try-catch here to catch a NetworkException. Ie if the server timeout.

Reply:
Caution: HttpClient's responses are super-lazy. Any exceptions would be thrown by the `toArray()` call below.

Reply:
Yes, sorry. 

@derrabus are correct. Add the try-catch around the `toArray()`.

Reply:
The other bridges use HTTP status code check, so I did the same for the sake of consistency.

Reply:
If there is a NetworkException it means that the request never reached the server. You will not have a HTTP status code then. 

Please add the try/catch. 

I know some other bridges don't do this, they should be updated. (In a separate PR)

Reply:
Just in case: this won't break the error checking logic below. Their API is a little weird and returns 200 even if there was an error. Doesn't make much sense to me, but that's the way they do it

Reply:
Done. Intercepted all of them via a common interface. Unless it's 200 and we can parse JSON there are no error details anyway

Reply:
Thank you. 

>Their API is a little weird and returns 200 even if there was an error.

Hm.. indeed. =)